domain/machine/state/state.go

```diff
@@ -81,6 +82,7 @@ VALUES ($M.machine_uuid, $M.net_node_uuid, $M.machine_id, $M.life_id)
 		if err != nil {
 			return errors.Trace(err)
 		}
+		newUUID = machineUUID.String()
```

Comment:
We should be doing this up in the service layer. If retries are invoked, we're generating them over again.